MapboxNavigation/NavigationMapView.swift

```diff
@@ -1350,3 +1389,130 @@ extension NavigationMapView {
     }
 }
 
+// MARK: - Building Extrusion Highlights
+
+private struct BuildingHighlightAttributes {
+    var location: CLLocationCoordinate2D = kCLLocationCoordinate2DInvalid
+    var highlightColor: UIColor = .defaultBuildingHighlightColor
+    var identifier: UInt64? = nil
+}
+
+extension NavigationMapView {
+       
+    /**
+     Receives coordinates for searching the map for buildings. If buildings are found, they will be highlighted in 2d or 3d depending on the `in3D` value.
+     `in3D` defaults to true
+     `buildingHighlightingEnabled` must be set to true before this function will work.
+     */
+    public func highlightBuildings(for coordinates: [CLLocationCoordinate2D], in3D: Bool = true) {
+        let attributes = coordinates.map { BuildingHighlightAttributes(location: $0, highlightColor: buildingHighlightColor) }
+        highlightBuildings(with: attributes, in3D: in3D)
+    }
+    
+    /**
+     Removes the highlight from all buildings highlighted by `highlightBuildings(for coordinates:, in3D:)`.
+     */
+    public func unhighlightBuildings() {
+        highlightBuildings(nil, in3D: false, extrudeAll: false)
+    }
+    
+    private func highlightBuildings(with attributes: [BuildingHighlightAttributes], in3D: Bool) {
+        let buildingAttributes = attributes.map {(value) -> BuildingHighlightAttributes in
+            var output = value
```

Comment:
I guess that return result is not really used anywhere. Maybe it'd be better to simplify current logic to stop using `BuildingHighlightAttributes` for now? We can handle edge cases like rendering highlighted building only when they appear on screen as a tail work.